addons/account/tests/test_xml_verifier.py

```diff
@@ -0,0 +1,93 @@
+
+
+@tagged('standard', 'at_install')
+class XMLVerifierTest(TransactionCase):
+
+    @classmethod
+    def setUpClass(cls):
+        res = super().setUpClass()
+        cls.XMLSIG_NSMAP = {"ds": "http://www.w3.org/2000/09/xmldsig#", "xades": "http://uri.etsi.org/01903/v1.3.2#"}
+
+        with file_open("account/tools/xml_verifier/xades_signed.xml", "rt") as f:
+            cls.xades_sample = lxml.etree.fromstring(f.read().encode())
+
+        cls.signature_node = cls.xades_sample.find('ds:Signature', namespaces=cls.XMLSIG_NSMAP)
+        return res
+
+    def test_xmldsig_verification(self):
+        try:
+            validate_xmldsig_signature(self.signature_node)
+            validation_successful = True
+        except Exception:
+            validation_successful = False
+
+        self.assertTrue(validation_successful)
+
+        # Invalid reference digest value.
+        signature_with_wrong_digest = deepcopy(self.xades_sample).find('ds:Signature', namespaces=self.XMLSIG_NSMAP)
+        signature_with_wrong_digest.find(".//ds:DigestValue", namespaces=self.XMLSIG_NSMAP).text = base64.b64encode(str.encode("Wrong Digest"))
+        with self.assertRaises(ValueError, msg="Should throw a ValueError when the references are not valid"):
+            validate_xmldsig_signature(signature_with_wrong_digest)
+
+        # Signature method not found.
+        signature_with_invalid_method = deepcopy(self.xades_sample).find('ds:Signature', namespaces=self.XMLSIG_NSMAP)
+        signature_with_invalid_method.find(".//ds:SignatureMethod", namespaces=self.XMLSIG_NSMAP).set("Algorithm", "New signature")
+        with self.assertRaises(ValueError, msg="Should throw a ValueError when the signature method is not found"):
+            validate_xmldsig_signature(signature_with_invalid_method)
+
+        # Reference URI not found.
+        signature_with_invalid_uri = deepcopy(self.xades_sample).find('ds:Signature', namespaces=self.XMLSIG_NSMAP)
+        signature_with_invalid_uri.findall(".//ds:Reference", namespaces=self.XMLSIG_NSMAP)[1].set("URI", "#Invalid-URI")
+        with self.assertRaises(ValueError, msg="Should throw a ValueError when reference URI is not found"):
+            validate_xmldsig_signature(signature_with_invalid_uri)
+
+        # Invalid signature.
+        signature_invalid = deepcopy(self.xades_sample).find('ds:Signature', namespaces=self.XMLSIG_NSMAP)
+        signature_invalid.find("ds:SignatureValue", namespaces=self.XMLSIG_NSMAP).text = base64.b64encode(str.encode("Wrong Signature"))
+        with self.assertRaises(InvalidSignature, msg="Should throw an InvalidSignature when signature is invalid"):
+            validate_xmldsig_signature(signature_invalid)
+
+    def test_xades_verification(self):
+        try:
+            validate_xades_signature(self.signature_node)
+            validation_successful = True
+        except Exception:
+            validation_successful = False
+
+        self.assertTrue(validation_successful)
```

Comment:
```suggestion
         validate_xades_signature(self.signature_node)
```
If it raise, it fails

Reply:
Okay, that's doable in this case, but what about in `account_move.py` where I need to validate the signature? At the moment I'm doing `except Exception` because I need the error message passed from the validate function. What can I do in this case?

Reply:
Because I missed it